diff --git a/c++/g2.cpp b/c++/g2.cpp
index 0bed562..8ffe76b 100644
--- a/c++/g2.cpp
+++ b/c++/g2.cpp
@@ -177,7 +177,7 @@ namespace pomerol2triqs {
     else
       g2 = compute_g2<w_nu_nup_t>(p.gf_struct, mesh_bff, p.block_order, p.blocks, filler);
 
-    g2() = mpi_all_reduce(g2(), comm);
+    g2() = mpi::all_reduce(g2(), comm);
 
     return g2;
   }
@@ -250,7 +250,7 @@ namespace pomerol2triqs {
     };
 
     auto g2 = compute_g2<w_l_lp_t>(p.gf_struct, mesh, p.block_order, p.blocks, filler);
-    g2() = mpi_all_reduce(g2(), comm);
+    g2() = mpi::all_reduce(g2(), comm);
 
     return g2;
   }
diff --git a/c++/pomerol_ed.hpp b/c++/pomerol_ed.hpp
index 35f5b99..0776eee 100644
--- a/c++/pomerol_ed.hpp
+++ b/c++/pomerol_ed.hpp
@@ -30,7 +30,8 @@
 #include <triqs/gfs.hpp>
 #include <triqs/operators/many_body_operator.hpp>
 #include <triqs/hilbert_space/fundamental_operator_set.hpp>
-#include <triqs/mpi/boost.hpp>
+//#include <triqs/mpi/boost.hpp>
+#include <boost/mpi.hpp>
 
 #include "g2_parameters.hpp"
 
@@ -58,7 +59,7 @@ namespace pomerol2triqs {
   /// Main solver class of pomerol2triqs
   class pomerol_ed {
 
-    triqs::mpi::communicator comm;
+    boost::mpi::communicator comm;
 
     const bool verbose;
     index_converter_t index_converter;
